### PR TITLE
Add support for Facebook sign up and login

### PR DESCRIPTION
**Pull request by MathiasLantean**

### CODE REVIEW

File: base/requirements/base.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+idna==2.9
+oauthlib==3.1.0
+python-dotenv==0.12.0
+python3-openid==3.1.0
+pytz==2019.3
+requests==2.23.0
+requests-oauthlib==1.3.0
+sqlparse==0.3.1
+urllib3==1.25.8
+UserManager==0.5.3
```
Comment (kaozdl) [inline base/requirements/base.txt:19]: we should consider switching to pipenv in the future

File: base/settings/develop.py [hunk 2]
```diff
@@ -14,6 +14,6 @@
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, '../../db.sqlite3'),
+        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
```
Comment (kaozdl) [inline base/settings/develop.py:17]: also, for further functionalities you will need to use postgresql and postgis

File: profile/serializers.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+from django.conf import settings
+from rest_framework import exceptions, serializers
+from allauth.utils import get_username_max_length, email_address_exists
+from allauth.account.adapter import get_adapter
+from allauth.account.utils import setup_user_email
+from allauth.account import app_settings as allauth_settings
+from dj_rest_auth.serializers import LoginSerializer as dj_rest_auth_LoginSerializer
+from .models import User
+
+UserModel = get_user_model()
```
Comment (kaozdl) [inline profile/serializers.py]: why is this line here? it doesn't seem to be used elsewhere

File: templates/account/email_confirm.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+{% else %}
+
+{% url 'account_email' as email_url %}
+
+<p>{% blocktrans %}This e-mail confirmation link expired or is invalid. Please <a href="{{ email_url }}">issue a new e-mail confirmation request</a>.{% endblocktrans %}</p>
+
+{% endif %}
+
+{% endblock %}
```
Comment (kaozdl) [inline templates/account/email_confirm.html]: Missing newline

File: templates/account/facebook_sign_up.html [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        }
+
+        function statusChangeCallback(response) {
+            document.getElementById("fb-key").innerHTML = response.authResponse.accessToken;
+        }
+
+    </script>
+
+</body>
+</html>
```
Comment (kaozdl) [inline templates/account/facebook_sign_up.html]: Missing newline

File: base/settings/develop.py (hunk 2 above)
Comment (MathiasLantean) [reply base/settings/develop.py:17]: Is it okay if I deal with it at that point or do you want me to do it for this task?

File: profile/serializers.py (hunk 3 above)
Comment (MathiasLantean) [reply profile/serializers.py]: After testing it, this line has nothing to do here.

File: profile/views.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+from allauth.socialaccount.providers.facebook.views import FacebookOAuth2Adapter
+from dj_rest_auth.registration.views import SocialLoginView
+from django.contrib.auth import get_user_model
+from django.shortcuts import render
+from rest_framework import viewsets
+from .serializers import UserSerializer
+User = get_user_model()
```
Comment (guilleijo) [inline profile/views.py:7]: It looks like you don't need some of these imports

**Review by guilleijo — APPROVED** [review]
Left a comment

**Review by kaozdl — APPROVED** [review]
